.github/workflows/codeql.yml

```diff
@@ -0,0 +1,36 @@
+name: "Static Analysis: CodeQL Scan"
+
+on:
+  push:
+    branches:
+      - "pull-request/[0-9]+"
+      - "main"
+
+jobs:
+  analyze:
+    name: Analyze (${{ matrix.language }})
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - language: python
+          build-mode: none
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
```

Comment:
Note I didn't pin these to specific commits like I did for Bandit above because I figured that these are GitHub managed where we'd be more secure allowing to float forward to get fixes than pinning to a specific commit. I figure if GitHub gets compromised we're in trouble regardless 😄.

Reply:
Either way is fine. I dunno if we can set up a dependentbot, if we can then it supports either pinning.

Reply:
We probably should set up dependabot or something similar in general